gateway/ln-gateway/src/client.rs

```diff
@@ -36,19 +41,34 @@ impl GatewayClientBuilder {
         }
     }
 
-    pub async fn build(
+    async fn client_plainrootsecret(&self, db: &Database) -> Result<DerivableSecret> {
+        let client_secret =
+            if let Ok(secret) = Client::load_decodable_client_secret::<[u8; 64]>(db).await {
+                secret
+            } else {
+                info!("Generating secret and writing to client storage");
+                let secret = PlainRootSecretStrategy::random(&mut thread_rng());
+                Client::store_encodable_client_secret(db, secret)
+                    .await
+                    .map_err(GatewayError::ClientStateMachineError)?;
+                secret
```

Comment:
Do we still need this branch in `client_plainrootsecret`? I'd imagine all new clients use the global secret, which is a BIP39 one and not plain?

Reply:
Correct, not necessary, good catch. Removed.